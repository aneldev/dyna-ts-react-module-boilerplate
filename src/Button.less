.my-button {
  outline: none;

  &--style- {

    &ROUNDED {
      font-size: 0;

      button {
        outline: none;
        border-style: solid;
        cursor: pointer;
        transition: background-color 200ms ease-out;
      }

      &.my-button--size- {
        &SMALL {
          button {
            padding: 2px 8px;
            font-size: 8px;
            line-height: 10px;
            border-width: 1px;
            border-radius: 8px;
            font-weight: bold;
            &:active {
              position: relative;
              top: 1px;
              left: 1px;
            }
          }
        }
        &MEDIUM {
          // line-height: 8px;
          button {
            padding: 4px 16px;
            font-size: 14px;
            line-height: 22px;
            border-width: 1px;
            border-radius: 16px;
            font-weight: bold;
            &:active {
              position: relative;
              top: 1px;
              left: 1px;
            }
          }
        }
        &LARGE {
          button {
            padding: 8px 32px;
            font-size: 26px;
            line-height: 46px;
            border-width: 1px;
            border-radius: 32px;
            font-weight: bold;
            &:active {
              position: relative;
              top: 2px;
              left: 2px;
            }
          }
        }
        &XLARGE {
          button {
            padding: 16px 64px;
            font-size: 40px;
            line-height: 92px;
            border-width: 2px;
            border-radius: 64px;
            font-weight: bold;
            &:active {
              position: relative;
              top: 2px;
              left: 2px;
            }
          }
        }
      }
    }

  }

  &--color- {
    &WHITE_BLACK {
      button {
        border-color: black;
        background: white;
        color: black;
        &:hover {
          background-color: #e6e6e6;
        }
        &:active {
          background-color: #d1d1d1;
        }
      }
    }
    &WHITE_RED {
      button {
        border-color: red;
        background: white;
        color: red;
        &:hover {
          background-color: #e6e6e6;
        }
        &:active {
          background-color: #d1d1d1;
        }
      }
    }
    &BLACK_WHITE {
      button {
        border-color: black;
        background: black;
        color: white;
        &:hover {
          background-color: #333333;
        }
        &:active {
          background-color: #525252;
        }
      }
    }
    &TRANSPARENT_WHITE {
      button {
        border-color: white;
        background: transparent;
        color: white;
        &:hover {
          border-color: #e6e6e6;
          color: #e6e6e6;
        }
        &:active {
          border-color: #d1d1d1;
          color: #d1d1d1;
        }
      }
    }

  }
}
