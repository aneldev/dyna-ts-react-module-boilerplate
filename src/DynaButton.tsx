import * as React from "react";

import "./dyna-button.less";

export enum EStyle {
  ROUNDED = "ROUNDED",
}

export enum EColor {
  WHITE_BLACK = "WHITE_BLACK",
  WHITE_RED = "WHITE_RED",
  BLACK_WHITE = "BLACK_WHITE",
  TRANSPARENT_WHITE = "TRANSPARENT_WHITE",
}

export enum ESize {
  SMALL = "SMALL",
  MEDIUM = "MEDIUM",
  LARGE = "LARGE",
  XLARGE = "XLARGE",
}

export interface IDynaButtonProps {
  children: any;
  style?: EStyle;
  color?: EColor;
  size?: ESize;
  href?: string
  onClick?: () => void;
}

export class DynaButton extends React.Component<IDynaButtonProps> {
  static defaultProps: IDynaButtonProps = {
    children: null,
    style: EStyle.ROUNDED,
    color: EColor.WHITE_BLACK,
    size: ESize.MEDIUM,
    href: null,
    onClick: () => undefined,
  };

  public render(): JSX.Element {
    const {
      children, style, color, size, href, onClick
    } = this.props;

    const className: string = [
      'dyna-button',
      `dnbtn-style-${style}`,
      `dnbtn-color-${color}`,
      `dnbtn-size-${size}`,
    ].join(' ').trim();

    return (
      <a className={className} href={href} onClick={onClick}>
        <button>{children}</button>
      </a>
    );
  }
}
