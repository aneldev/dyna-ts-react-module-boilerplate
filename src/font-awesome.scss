$fa-font-path: '~font-awesome/fonts';
@import "~font-awesome/scss/variables";
@import "~font-awesome/scss/mixins";
@import "~font-awesome/scss/core";
@import "~font-awesome/scss/icons";
