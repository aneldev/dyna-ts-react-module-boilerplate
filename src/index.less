.plainButton {
  font-family: sans-serif;
  display: inline;
  padding: 20px;
  color: white;
  font-size: 1.5rem;
  cursor: pointer;

  transition: color ease 500ms;

  &:active {
    position: relative;
    top: 1px;
    left: 1px;
  }

  &:focus {
    outline: none; // remove the outer glow of Chrome on focus
  }
}

.ok {
  color: white;
  background-color: dodgerblue;
  &:hover {
    color: lightgrey;
  }
}

.cancel {
  color: white;
  background-color: tomato;
  &:hover {
    color: lightgrey;
  }
}

.other {
  color: black;
  background-color: whitesmoke;
  &:hover {
    color: darkgray;
  }
}
