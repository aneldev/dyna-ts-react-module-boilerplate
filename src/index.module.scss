.plainButton {
  border: 1px solid lightgray;
}