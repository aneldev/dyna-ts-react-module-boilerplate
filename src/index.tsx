import * as React from 'react';

import {MouseEventHandler} from "react";

let  stylesFromLess: any = require("./index.less");
let  stylesFromScss: any = require("./index.scss");

export enum EButtonType {
  OK, CANCEL, OTHER
}

export interface IPlainButtonProps {
  caption: string;
  buttonType?: EButtonType;
  onClick?: MouseEventHandler<any>;
}
export interface IPlainButtonState {
}

export default
class PlainButton extends React.Component<IPlainButtonProps, IPlainButtonState> {
  public static defaultProps: IPlainButtonProps = {
    caption: '',
    buttonType: EButtonType.OTHER,
    onClick: () => undefined,
  };

  constructor(props: IPlainButtonProps, context: any) {
    super(props, context);
  }

  private get getTypeClassName():string{
    switch (this.props.buttonType){
      case EButtonType.OK: return stylesFromLess.ok;
      case EButtonType.CANCEL: return stylesFromLess.cancel;
      case EButtonType.OTHER: return stylesFromLess.other;
    }
  }

  public render(): JSX.Element {
    // ReactDOM;
    let className:string = `${stylesFromLess.plainButton} ${stylesFromScss.plainButton} ${this.getTypeClassName}`;
    return (
      <button className={className} onClick={this.props.onClick}>
        {this.props.caption}
      </button>
    );
  }

}

