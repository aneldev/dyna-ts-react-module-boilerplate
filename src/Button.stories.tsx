import * as React from 'react';
import {Button} from './Button';

export default {title: 'My Button'};

export const basic = () => <Button/>;
