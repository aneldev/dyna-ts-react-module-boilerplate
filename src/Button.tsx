import * as React      from "react";
import {dynaClassName} from "dyna-class-name";

import "./Button.less";

export enum EStyle {
  ROUNDED = "ROUNDED",
}

export enum EColor {
  WHITE_BLACK = "WHITE_BLACK",
  WHITE_RED = "WHITE_RED",
  BLACK_WHITE = "BLACK_WHITE",
  TRANSPARENT_WHITE = "TRANSPARENT_WHITE",
}

export enum ESize {
  SMALL = "SMALL",
  MEDIUM = "MEDIUM",
  LARGE = "LARGE",
  XLARGE = "XLARGE",
}

export interface IButtonProps {
  children: any;
  style?: EStyle;
  color?: EColor;
  size?: ESize;
  href?: string;
  onClick?: () => void;
}

export class Button extends React.Component<IButtonProps> {
  static defaultProps: IButtonProps = {
    children: null,
    style: EStyle.ROUNDED,
    color: EColor.WHITE_BLACK,
    size: ESize.MEDIUM,
    href: undefined,
    onClick: () => undefined,
  };

  private readonly cn = dynaClassName("my-button");

  public render(): JSX.Element {
    const {
      children, style, color, size, href, onClick
    } = this.props;

    const className: string = `--style-${style} --color-${color} --size-${size}`;

    return (
      <a className={this.cn.root(this.props, className)} href={href} onClick={onClick}>
        <button>{children}</button>
      </a>
    );
  }
}
