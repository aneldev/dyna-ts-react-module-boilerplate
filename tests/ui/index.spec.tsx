declare let global: any, jasmine: any, describe: any, clearTest: any, it: any, expect: any;

import { configure } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import * as React from 'react';
import * as enzyme from 'enzyme';

import {DynaButton} from '../../src';

describe('Home', () => {
  let wrapper;

  it('has expected content with deep render', () => {
    wrapper = enzyme.shallow(
      (
        <DynaButton onClick={() => console.log('I am clicked')}>My button</DynaButton>
      ),
      {}
    );

    expect(wrapper).toMatchSnapshot();
  });
});
