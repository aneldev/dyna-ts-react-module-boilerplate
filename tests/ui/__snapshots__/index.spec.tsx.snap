// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home has expected content with deep render 1`] = `
<a
  className="my-button my-button--style-ROUNDED my-button--color-WHITE_BLACK my-button--size-MEDIUM"
  href={null}
  onClick={[Function]}
>
  <button>
    My button
  </button>
</a>
`;
